apollo-router/src/main.rs

```diff
@@ -134,22 +134,39 @@ async fn rt_main() -> Result<()> {
         })
         .unwrap_or_else(|| ConfigurationKind::Instance(Configuration::builder().build().boxed()));
 
-    let schema = if let Ok(apollo_key) = std::env::var("APOLLO_KEY") {
-        let apollo_graph_ref = std::env::var("APOLLO_GRAPH_REF")
-            .expect("cannot set up Uplink schema download if  the APOLLO_GRAPH_REF environment variable is not set");
+    let schema = match (opt.supergraph_path, std::env::var("APOLLO_KEY")) {
+        (Some(supergraph_path), _) => {
+            let supergraph_path = if supergraph_path.is_relative() {
+                current_directory.join(supergraph_path)
+            } else {
+                supergraph_path
+            };
+            SchemaKind::File {
+                path: supergraph_path,
+                watch: opt.watch,
+                delay: None,
+            }
+        }
+        (None, Ok(apollo_key)) => {
+            let apollo_graph_ref = std::env::var("APOLLO_GRAPH_REF")
+            .map_err(|_| anyhow!("cannot set up Uplink schema download if the APOLLO_GRAPH_REF environment variable is not set"))?;
 
-        SchemaKind::Registry {
-            apollo_key,
-            apollo_graph_ref,
+            SchemaKind::Registry {
+                apollo_key,
+                apollo_graph_ref,
+            }
         }
-    } else {
-        ensure!(
-            opt.supergraph_path.is_some(),
-            r#"
+        _ => {
+            return Err(anyhow!(
+                r#"
     💫 Apollo Router requires a supergraph to be set using '--supergraph':
 
         $ ./router --supergraph <file>`
 
+        Or using an Uplink by setting the APOLLO_KEY and APOLLO_GRAPH_REF environment variable
+
+        $ APOLLO_KEY=XXX APOLLO_GRAPH_REF=XXX ./router
+
```

Comment:
The term "Uplink" isn't really socialized enough yet so probably best to say "Managed Federation" here

```suggestion
        Alternatively, to retrieve the supergraph from Apollo Studio, set the APOLLO_KEY
        and APOLLO_GRAPH_REF environment variables to your graph's settings.
        
          $ APOLLO_KEY="..." APOLLO_GRAPH_REF="..." ./router
          
        For more on Apollo Studio and Managed Federation, see our documentation:
        
          https://www.apollographql.com/docs/router/managed-federation/

```